themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,196 @@
+  id: string;
+  name: string;
+}
+
+export interface Consent {
+  createDate: number;
+  grantedScopes: GrantedScope[];
+  lastUpdatedDate: number;
+}
+
+interface Application {
+  baseUrl: string;
+  clientId: string;
+  clientName: string;
+  consent: Consent;
+  description: string;
+  inUse: boolean;
+  offlineAccess: boolean;
+  userConsentRequired: boolean;
+  scope: string[];
+}
+
+export class ApplicationsPage extends React.Component<ApplicationsPageProps, ApplicationsPageState> {
+
+  public constructor(props: ApplicationsPageProps) {
+    super(props);
+    this.state = {
+      isRowOpen: [],
+      applications: []
+    };
+
+    this.fetchApplications();
+  }
+
+  private removeConsent = (clientId: string) => {
+    AccountServiceClient.Instance.doDelete("/applications/" + clientId + "/consent")
+      .then(() => {
+        this.fetchApplications();
+      });
+  }
+
+  private onToggle = (row: number): void => {
+    const newIsRowOpen: boolean[] = this.state.isRowOpen;
+    newIsRowOpen[row] = !newIsRowOpen[row];
+    this.setState({ isRowOpen: newIsRowOpen });
+  };
+
+  private fetchApplications(): void {
+    AccountServiceClient.Instance.doGet("/applications")
+      .then((response: AxiosResponse<Object>) => {
+        const applications = response.data as Application[];
+        this.setState({
+          isRowOpen: this.collapseRows(applications),
+          applications: applications
+        });
+      });
+  }
+
+  private collapseRows(applications: Application[]): boolean[] {
+    const openRows: boolean[] = new Array<boolean>().fill(true);
+    applications.forEach((application: Application, appIndex: number) => {
+      openRows[appIndex] = false;
+    });
+    return openRows;
+  }
+
+  public render(): React.ReactNode {
+    return (
+      <ContentPage title="applications">
+        <DataList id="applications-list" aria-label="Applications">
+          {this.state.applications.map((application: Application, appIndex: number) => {
+            return (
+              <DataListItem key={'application-' + appIndex} aria-labelledby="applications-list" isExpanded={this.state.isRowOpen[appIndex]}>
+                <DataListItemRow>
+                  <DataListToggle
+                    onClick={() => this.onToggle(appIndex)}
+                    isExpanded={this.state.isRowOpen[appIndex]}
+                    id={'applicationToggle' + appIndex}
+                    aria-controls="expandable"
+                  />
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell width={2} key={'app-' + appIndex}>
+                        <BuilderImageIcon /> <Text component={TextVariants.small}>{application.clientId}</Text>
+                      </DataListCell>,
+                      <DataListCell width={2} key={'internal-' + appIndex}>
+                        <span>{application.inUse && !application.userConsentRequired ? 'Internal' : 'Third-party'}
```

Comment:
I believe we don't care about `application.inUse` when deciding whether an app is internal or 3rd-party.

Reply:
@vmuzikar I tried to keep things in sync with all the requirements requested by @stianst, but you're correct. We don't need that.